### PR TITLE
[AMORO-3131]  Add the grafana dashboard template

### LINKED ISSUE #3131: [Improvement]: OOTB grafana dashbord json
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

We got an excellent feature of prometheus metrics exporter in version 0.7, and further more, we can give user more convenient tools for out-of-the-box monitor dashboard such as template of grafana.

### How should we improve?

As mentioned below, maybe a grafana json file or template id is a good ways for user to build a dashboard.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by jiamin13579**
## Why are the changes needed?

Close #3131.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: grafana/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+3. **Import Dashboard**:
+    - Go to the "+" menu, select "Import".
+    - Upload the JSON file or paste its contents.
+4. **Configure Data Sources**: Map any required data sources to your existing ones.
+5. **Save**: Confirm and save the dashboard.
+
+## Notes
+
+- Ensure data sources are properly configured.
+- Customize variables, thresholds, and layouts as needed.
```
Comment (tcodehuber) [inline grafana/README.md:20]: Can you add more details how to configure amoro metric reporter？ Or put a href link to official website.

**Review by tcodehuber — COMMENTED** [review]
Thanks for your contribution. I have one comment, PTAL.

File: grafana/README.md (hunk 1 above)
Comment (jiamin13579) [reply grafana/README.md]: Ok, I have added a link to official website.

File: grafana/dashboard.json [hunk 2]
```diff
@@ -0,0 +1,2407 @@
...
+          "editorMode": "code",
+          "exemplar": false,
+          "expr": "amoro_table_optimizing_process_total_count_total{catalog=~\"$Catalog\", database=~\"$Database\", table=~\"$Table\"} > 0",
+          "format": "time_series",
+          "fullMetaSearch": false,
+          "hide": false,
+          "includeNullMetadata": true,
+          "instant": false,
+          "interval": "",
+          "legendFormat": "{{database}}.{{table}}",
```
Comment (huyuanfeng2018) [inline grafana/dashboard.json:1458]: {{catalog}}.{{database}}.{{table}}?
Is the catalog missing?

File: grafana/dashboard.json [hunk 3]
```diff
@@ -0,0 +1,2407 @@
...
+        "overrides": []
+      },
+      "gridPos": {
+        "h": 8,
+        "w": 12,
+        "x": 0,
+        "y": 45
+      },
+      "id": 34,
+      "maxDataPoints": 1,
```
Comment (huyuanfeng2018) [inline grafana/dashboard.json:1632]: maxDataPoints=1，There may be problems when the interval is large. Here it looks like this

**Review by huyuanfeng2018 — COMMENTED** [review]
@jiamin13579 Thank you very much for contribution.
Looks really good, I left some comments, PTAL

File: grafana/dashboard.json (hunk 2 above)
Comment (jiamin13579) [reply grafana/dashboard.json:1458]: I thought that adding catalog would make the name very long, so I didn't add it. What suggestions do you have?

image

File: grafana/dashboard.json (hunk 3 above)
Comment (jiamin13579) [reply grafana/dashboard.json]: maxDataPoints has been adjusted to default.

**Review by huyuanfeng2018 — APPROVED** [review]
LGTM.

File: grafana/dashboard.json [hunk 4]
```diff
@@ -0,0 +1,2582 @@
...
+      ],
+      "title": "Top Total Count",
+      "transformations": [
+        {
+          "id": "reduce",
+          "options": {
+            "includeTimeField": false,
+            "mode": "seriesToRows",
+            "reducers": [
+              "lastNotNull"
```
Comment (huyuanfeng2018) [inline grafana/dashboard.json:1662]: How about using `difference` here?

Comment (jiamin13579) [reply grafana/dashboard.json:1665]: Great suggestion, I have updated it!

**Review by XBaith — APPROVED** [review]
LGTM

**Pull request merged**